build/publishNewVersion.mjs

```diff
@@ -0,0 +1,117 @@
+    .option('--force <packages...>', 'force a bump on specified package', ['root'])
+    .addOption(new Option('--bump <type>', 'bump a <type> version instead of reliying on commit messages').choices(BUMP_TYPES))
+
+program.parse();
+
+const options = program.opts();
+
+// We consider that root always change, we always want to bump a version
+if (!options.force.includes('root')){
+    options.force.push('root');
+}
+
+(async () => {
+    const convention = await angularChangelogConvention;
+
+    const lastTag = getLastTag(VERSION_PREFIX);
+    console.log('Last tag: %s', lastTag);
+
+    const remote = getRemoteName();
+    const since = [remote, lastTag].filter(Boolean).join('/');
+
+    console.log('Since: %s', since);
+
+    const changedPackages = pnpmGetChangedPackages(since);
+    options.force.forEach(packageName => {
+        if (!changedPackages.includes(packageName)) {
+            changedPackages.push(packageName);
+        }
+    });
+
+    if (changedPackages.length > 0) {
+        const commits = getCommits(PATH, lastTag);
+
+        const parsedCommits = parseCommits(commits, convention.parserOpts);
+        let bumpInfo;
+        if (options.bump !== null && BUMP_TYPES.includes(options.bump)) {
+            bumpInfo = {type: options.bump, preid: options.bump === 'prerelease' ? 'next' : undefined};
+        } else {
+            bumpInfo = convention.recommendedBumpOpts.whatBump(parsedCommits);
+        }
+
+        const currentVersion = getCurrentVersion(PATH);
+        const newVersion = getNextVersion(currentVersion, bumpInfo);
+
+        if (newVersion !== currentVersion) {
+            console.log('Bumping %s to version %s', changedPackages.join(', '), newVersion)
+            pnpmBumpVersion(newVersion, since, options.force);
+
+            if (parsedCommits.length > 0) {
+                const changelog = await generateChangelog(
+                    parsedCommits,
+                    newVersion,
+                    {
+                        host: 'https://github.com',
+                        owner: 'coveo',
+                        repository: 'plasma',
+                    },
+                    convention.writerOpts
+                );
+                await writeChangelog(PATH, changelog);
+            }
+
+            const versionTag = `${VERSION_PREFIX}${newVersion}`;
+            if (!options.dry) {
+                gitCommit(`[version bump]: publish version ${versionTag}`, '.');
+                gitTag(versionTag);
+                if (remote) {
+                    gitPush();
+                    gitPushTags();
+
+                    const forcePackagesWithoutRoot = options.force.filter(packageName => packageName !== 'root');
+                    pnpmPublish(since, options.tag, options.branch, forcePackagesWithoutRoot, ['root']);
+                }
+            }
+
+            // const octokit = new Octokit({auth: process.env.GITHUB_CREDENTIALS});
+            // await createGitHubRelease(
```

Comment:
WIP?

Reply:
Yes, I'll make sure the rest work before trying to create a release (we don't have those anymore)